### PR TITLE
Additional mass reco variables and (DNN applied but not used) 

### PR DESCRIPTION
**Pull request by tihsu99**

### CODE REVIEW

File: BDT_training/step1_Samplepreparation/runCondor.py [hunk 1]
```diff
@@ -35,11 +36,19 @@ def prepare_shell(shell_file, command, condor, FarmDir):
...
       Eras.append(Era)
 
-  cmsswBase = os.environ['CMSSW_BASE']
+  cwd = os.getcwd()
+  dir_list = cwd.split('/')
+  cmssw_list = []
+  for dir_ in dir_list:
+    cmssw_list.append(dir_)
+    if 'CMSSW' in dir_: break
+  cmsswBase = '/'.join(cmssw_list)
```
Comment (gourangakole) [inline BDT_training/step1_Samplepreparation/runCondor.py:45]: Why do you need this? 

old cmsswBase = os.environ['CMSSW_BASE'] should give same results?

File: BDT_application/step1_Samplepreparation/runCondor.py [hunk 2]
```diff
@@ -36,14 +36,14 @@ def prepare_shell(shell_file, command, condor, FarmDir):
...
       Eras.append(Era)
 
-  cmsswBase = os.environ['CMSSW_BASE']
-  FarmDir   = '%s/Farm_BDT'%cmsswBase
   cwd       = os.getcwd()
-  os.system('mkdir -p %s'%FarmDir)
   os.system('cp %s/../../python/common.py .'%cwd)
 
-  from common import inputFile_path
+  from common import inputFile_path, cmsswBase
```
Comment (gourangakole) [inline BDT_application/step1_Samplepreparation/runCondor.py:42]: Hi @AlbertHsuNTUphys why do you need extra function for ``cmsswBase``? 
I added a similar comment on common.py file.

File: BDT_training/step1_Samplepreparation/runCondor.py [hunk 3]
```diff
@@ -49,7 +58,7 @@ def prepare_shell(shell_file, command, condor, FarmDir):
   condor.write('log    = %s/job_common.log\n'%FarmDir)
   condor.write('executable = %s/$(cfgFile)\n'%FarmDir)
   condor.write('requirements = (OpSysAndVer =?= "CentOS7")\n')
-  condor.write('+JobFlavour = "tomorrow"\n')
+  condor.write('+JobFlavour = "longlunch"\n')
```
Comment (gourangakole) [inline BDT_training/step1_Samplepreparation/runCondor.py]: just a double check
does ``longlunch`` finished all job in few hours?

File: data/DNN_Flav_MET_threeJet_model_v2.dat
Comment (gourangakole) [inline data/DNN_Flav_MET_threeJet_model_v2.dat:1]: Just for my understanding, I hope this file is automatically written by some function, but which function?

File: data/DNN_Flav_MET_threeJet_model_v2.hxx
Comment (gourangakole) [inline data/DNN_Flav_MET_threeJet_model_v2.hxx:1]: same as the above DNN command.

File: data/DNN_Flav_MET_threeJet_model_v2.onnx
Comment (gourangakole) [inline data/DNN_Flav_MET_threeJet_model_v2.onnx:1]: same as the above DNN command.

File: data/DNN_woBDTinput.dat
Comment (gourangakole) [inline data/DNN_woBDTinput.dat:1]: same as the above DNN command.

File: data/DNN_woBDTinput.hxx
Comment (gourangakole) [inline data/DNN_woBDTinput.hxx:1]: same as the above DNN command.

File: python/common.py [hunk 4]
```diff
@@ -8,7 +8,16 @@
...
+
+cwd = os.getcwd()
+dir_list = cwd.split('/')
+cmssw_list = []
+for dir_ in dir_list:
+  cmssw_list.append(dir_)
+  if 'CMSSW' in dir_: break
+cmsswBase = '/'.join(cmssw_list)
+
+#cmsswBase = os.environ['CMSSW_BASE']
```
Comment (gourangakole) [inline python/common.py:20]: old cmsswBase should be ok no?

File: BDT_training/step1_Samplepreparation/runCondor.py (hunk 1 above)
Comment (tihsu99) [reply BDT_training/step1_Samplepreparation/runCondor.py:45]: New DNN application used ROOT v6.28, which is not provided by any CMSSW now. Need new way to obtain this.

File: BDT_training/step1_Samplepreparation/runCondor.py (hunk 3 above)
Comment (tihsu99) [reply BDT_training/step1_Samplepreparation/runCondor.py]: For BDT training, it finish most of the time, but I can bring it back to tomorrow to be more safe.

File: data/DNN_Flav_MET_threeJet_model_v2.dat
Comment (tihsu99) [reply data/DNN_Flav_MET_threeJet_model_v2.dat:1]: They are produced automatically. I will update README to introduce how to produce the files.

Comment (gourangakole) [reply data/DNN_Flav_MET_threeJet_model_v2.dat:1]: OK that;s good.

File: BDT_training/step1_Samplepreparation/runCondor.py (hunk 3 above)
Comment (gourangakole) [reply BDT_training/step1_Samplepreparation/runCondor.py]: That's great.

**Comment by gourangakole** [comment]
thanks ..once you update README and bring back condor_q  to tomorrow then I am ready to merge.

File: README.md [hunk 5]
```diff
@@ -3,8 +3,16 @@
 ```
 cmsrel CMSSW_10_6_29
 cd CMSSW_10_6_29/src
-cmsenv
+# DO NOT run cmsenv
```
Comment (gourangakole) [inline README.md]: hi @AlbertHsuNTUphys but for normal work flow we needed ``cmsenv`` right? 

or should we move the 
```
source script/env.sh
```
to here for root env?

Comment (tihsu99) [reply README.md]: I am not sure where do we need cmsenv. We do not have any plot code in Script_For_MVA, so probably will not affect the result whether we do cmsenv or not.

Comment (gourangakole) [reply README.md]: Yes, but we need root so should we move 
source script/env.sh
to have a unique root version while we run independently. 

Agree @AlbertHsuNTUphys ?

**Comment by gourangakole** [comment]
Thanks look good to me unless @efeyazgan wants to check I am ready to merge :)

**Pull request merged**